micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java

```diff
@@ -0,0 +1,279 @@
+    }
+
+    /**
+     * Kafka {@link Consumer} metrics binder
+     *
+     * @param kafkaConsumer consumer instance to be instrumented
+     * @param tags          additional tags
+     */
+    public KafkaMetrics(Consumer<?, ?> kafkaConsumer, Iterable<Tag> tags) {
+        this(kafkaConsumer::metrics, tags);
+    }
+
+    /**
+     * Kafka {@link Consumer} metrics binder
+     *
+     * @param kafkaConsumer consumer instance to be instrumented
+     */
+    public KafkaMetrics(Consumer<?, ?> kafkaConsumer) {
+        this(kafkaConsumer::metrics);
+    }
+
+    /**
+     * {@link KafkaStreams} metrics binder
+     *
+     * @param kafkaStreams instance to be instrumented
+     * @param tags         additional tags
+     */
+    public KafkaMetrics(KafkaStreams kafkaStreams, Iterable<Tag> tags) {
+        this(kafkaStreams::metrics, tags);
+    }
+
+    /**
+     * {@link KafkaStreams} metrics binder
+     *
+     * @param kafkaStreams instance to be instrumented
+     */
+    public KafkaMetrics(KafkaStreams kafkaStreams) {
+        this(kafkaStreams::metrics);
+    }
+
+    /**
+     * Kafka {@link AdminClient} metrics binder
+     *
+     * @param adminClient instance to be instrumented
+     * @param tags        additional tags
+     */
+    public KafkaMetrics(AdminClient adminClient, Iterable<Tag> tags) {
+        this(adminClient::metrics, tags);
+    }
+
+    /**
+     * Kafka {@link AdminClient} metrics binder
+     *
+     * @param adminClient instance to be instrumented
+     */
+    public KafkaMetrics(AdminClient adminClient) {
+        this(adminClient::metrics);
+    }
+
+    KafkaMetrics(Supplier<Map<MetricName, ? extends Metric>> metricsSupplier) {
+        this(metricsSupplier, emptyList());
+    }
+
+    KafkaMetrics(Supplier<Map<MetricName, ? extends Metric>> metricsSupplier,
+            Iterable<Tag> extraTags) {
+        this.metricsSupplier = metricsSupplier;
+        this.extraTags = extraTags;
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        checkAndBindMetrics(registry);
+    }
+
+    /**
+     * Gather metrics from Kafka metrics API and register Meters.
+     * <p>
+     * As this is a one-off execution when binding a Kafka client, Meters include a call to this
+     * validation to double-check new metrics when returning values. This should only add the cost of
+     * validating meters registered counter when no new meters are present.
+     */
+    void checkAndBindMetrics(MeterRegistry registry) {
+        Map<MetricName, ? extends Metric> metrics = metricsSupplier.get();
+        //Only happens first time number of metrics change
+        if (currentSize.get() != metrics.size()) {
+            currentSize.set(metrics.size());
+            Map<String, Set<Meter>> boundMeters = new HashMap<>();
+            //Register meters
+            metrics.forEach((name, metric) -> {
+                //Filter out metrics from group "app-info", that includes metadata
```

Comment:
Found this KIP related to adding version and commit-id as metrics: https://cwiki.apache.org/confluence/display/KAFKA/KIP-188+-+Add+new+metrics+to+support+health+checks (#Client-side metrics)

And the implementation side: 

https://github.com/apache/kafka/blob/191f2faae07b6608b0601dc2caf204b196a8fc47/clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java#L89-L95

Reply:
Commit ID doesn't seem as useful unless snapshot versions are used, but maybe we could add the kafka client version as a tag to all the kafka metrics?

Reply:
Yes. I'll add `version` as tag, and `startTimeMs` as meter. Sounds good?

Reply:
`version` as a tag name can be problematic because it is likely to be used by a user as a common tag (possibly referring to the version of the application). Maybe a separate metric for the kafka client version is better, after all.